crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -209,6 +173,54 @@ impl Trade {
         })
     }
 
+    /// Protocol fee as a cut of surplus, denominated in SURPLUS token
+    fn surplus_fee(&self, factor: f64) -> Result<eth::Asset, Error> {
+        let surplus = self.surplus().ok_or(Error::Surplus(self.sell, self.buy))?;
+        let fee = surplus
+            .amount
+            .apply_factor(factor / (1.0 - factor))
+            .ok_or(Error::Factor(surplus.amount, factor))?;
+
+        Ok(eth::Asset {
+            token: surplus.token,
+            amount: fee,
+        })
+    }
+
+    /// Protocol fee as a cut of the trade volume, denominated in SURPLUS token
+    fn volume_fee(&self, factor: f64) -> Result<eth::Asset, Error> {
+        let executed_in_surplus_token: eth::TokenAmount = match self.side {
+            Side::Sell => self
+                .executed
+                .0
+                .checked_mul(self.custom_price.sell)
+                .ok_or(Math::Overflow)?
+                .checked_div(self.custom_price.buy)
+                .ok_or(Math::DivisionByZero)?,
+            Side::Buy => self
+                .executed
+                .0
+                .checked_mul(self.custom_price.buy)
+                .ok_or(Math::Overflow)?
+                .checked_div(self.custom_price.sell)
+                .ok_or(Math::DivisionByZero)?,
+        }
+        .into();
+        let factor = match self.side {
+            Side::Sell => factor / (1.0 - factor),
+            Side::Buy => factor / (1.0 + factor),
```

Comment:
Why this difference here? Comment in code please.

Reply:
resolved with added comment